spec/lib/workflow/stores/mongo-spec.js

```diff
@@ -619,32 +599,262 @@ describe('Task Graph mongo store interface', function () {
         });
     });
 
+    it('updates dependent tasks under anyOf key', function() {
+        var waterlineUnderAnyOf = {
+            taskdependencies:{
+                find: sinon.stub().resolves([]),
+                //make sure the find cannot find anything under dependent.taskId
+                updateMongo: sinon.stub().resolves()
+            }
+        };
+
+        helper.setupInjector([
```

Comment:
You don't have to do this in every test case. Just inject a stub waterline object, then you are free to change any of its properties.

```javascript
var sandbox = sinon.sandbox.create();
var mockWaterline = {
    taskdependencies: {
         find: null,
         updateMongo: null
    }
};

before(function() {
  helper.setupInjector([
    helper.di.simpleWrapper(mockWaterline, 'Services.Waterline')
  ]);
}

beforeEach(function() {
   mockWaterline.taskdependencies.find = sandbox.stub();
   mockWaterline.taskdependencies.updateMongo = sandbox.stub();
});

afterEach(function() {
   sandbox.restore();
});

it('xxxxxx', function() {
    mockWaterline.taskdependencies.find.resolves([]);
    mockWaterline.taskdependencies.updateMongo.resolves();
    .....
});

```